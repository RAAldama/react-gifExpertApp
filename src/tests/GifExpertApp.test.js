import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';

describe('Test on <GifExpertApp />', () => {
    
    test('should show <GifExpertApp /> correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show a list of categories', () => {
        
        const categories = ['Pikachu', 'Eevee'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();

        const gifGridQuantity = wrapper.find('GifGrid').length;
        expect(gifGridQuantity).toBe(categories.length);

    })
    
    
})
