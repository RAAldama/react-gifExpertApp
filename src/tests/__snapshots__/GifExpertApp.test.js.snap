// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifExpertApp /> should show <GifExpertApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test on <GifExpertApp /> should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Pikachu"
      key="Pikachu"
    />
    <GifGrid
      category="Eevee"
      key="Eevee"
    />
  </ol>
</Fragment>
`;
